### PR TITLE
Add integration test for max number of concurrent transfers

### PR DESCRIPTION
**Pull request by palango**
As requested by @taleldayekh

### CODE REVIEW

File: raiden/tests/integration/transfer/test_mediatedtransfer.py [hunk 1]
```diff
@@ -758,3 +761,77 @@ def assert_balances(expected_transferred_amounts=List[int]):
...
+        secrethash = sha256_secrethash(secret)
+
+        secrethashes.append(secrethash)
+
+        # Send payment while holding Unlock message
+        for shash in secrethashes:
+            hold_event_handler_app0.hold_unlock_for(secrethash=shash)
+            hold_event_handler_app1.hold_secretreveal_for(secrethash=shash)
+
+        failed_status = app0.mediated_transfer_async(
```
Comment (istankovic) [inline raiden/tests/integration/transfer/test_mediatedtransfer.py]: You don't use `failed_status` here inside the loop, but after the loop you reuse the name -- perhaps it'd be better to omit the assignment here?

File: raiden/tests/integration/transfer/test_mediatedtransfer.py [hunk 2]
```diff
@@ -758,3 +761,77 @@ def assert_balances(expected_transferred_amounts=List[int]):
...
+    for i in range(MAXIMUM_PENDING_TRANSFERS):
+        secret = Secret(keccak(i))
+        secrethash = sha256_secrethash(secret)
+
+        secrethashes.append(secrethash)
+
+        # Send payment while holding Unlock message
+        for shash in secrethashes:
+            hold_event_handler_app0.hold_unlock_for(secrethash=shash)
+            hold_event_handler_app1.hold_secretreveal_for(secrethash=shash)
```
Comment (istankovic) [inline raiden/tests/integration/transfer/test_mediatedtransfer.py]: Would it suffice to hold either one of those?

File: raiden/tests/integration/transfer/test_mediatedtransfer.py [hunk 3]
```diff
@@ -758,3 +761,77 @@ def assert_balances(expected_transferred_amounts=List[int]):
...
+        token_network_address=token_network_address,
+        amount=PaymentAmount(1),
+        target=TargetAddress(app1.address),
+        identifier=PaymentID(payment_id),
+        secret=secret,
+    )
+
+    failed_status.payment_done.wait()
+    result = failed_status.payment_done.value
+    assert type(result) == EventPaymentSentFailed
```
Comment (istankovic) [inline raiden/tests/integration/transfer/test_mediatedtransfer.py]: We should rather do direct type comparisons via `is`.

**Review by istankovic — APPROVED** [review]

File: raiden/tests/integration/transfer/test_mediatedtransfer.py (hunk 2 above)
Comment (palango) [reply raiden/tests/integration/transfer/test_mediatedtransfer.py]: In my understanding the unlock should be enough, but it isn't in practice. But I cannot explain why.

Comment (palango) [reply raiden/tests/integration/transfer/test_mediatedtransfer.py]: Mh, I thought I tested without the unlock, but actually it's not required.

Comment (achilles-portarinos) [reply raiden/tests/integration/transfer/test_mediatedtransfer.py]: (you might also want to update the comment at line 799),
btw, thanks for adding this test

Comment (palango) [reply raiden/tests/integration/transfer/test_mediatedtransfer.py]: @achilles-portarinos Good points, fixed it.

**Comment by ezdac** [comment]
The newly added integration test fails with:
```
E   RuntimeError: Matching event of type SendSecretReveal emitted twice, this should not happen. Either there is a bug in the state machine or the hold.attributes is too generic and multiple different events are matching. Event: SendSecretReveal(recipient=b'\xe1\x93\xfc\xca\x83J\xa9\xaa\x8eZXYc\xf9\xfeS\x9c\\\x83P', recipient_metadata={'user_id': '@0xe193fcca834aa9aa8e5a585963f9fe539c5c8350:localhost:8508', 'capabilities': 'mxc://raiden.network/cap?Receive=1&Mediate=1&Delivery=1&webRTC=1&toDevice=1&immutableMetadata=1', 'displayname': '0x02eb6583b4e0c00273859479e2663f9f47decc86e395ccaba57926b31a6aa2cf61e6d357ef1d274d1a0e06dc8d2efd728a2eefde033031c76102ef6bcadfa05b1b'}, canonical_identifier=CanonicalIdentifier(chain_identifier=0, token_network_address=b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', channel_identifier=0), message_identifier=5960386333235311427, secrethash=b'\xdb\x12\x90\x11\xf5oO\xc0\xb0\xc5\x0c\xa1\xff\xc8\xd7\xd1\xe1\xe8\xa7H?\xc2\x8f^\xf1\x801\x96\x9c%8;') Attributes: {'secrethash': b'\xdb\x12\x90\x11\xf5oO\xc0\xb0\xc5\x0c\xa1\xff\xc8\xd7\xd1\xe1\xe8\xa7H?\xc2\x8f^\xf1\x801\x96\x9c%8;'}
```

This seems to be a teardown related issue in the `HoldRaidenEventHandler` - I can reproduce this locally when running the whole `test_mediatedtransfer.py` integration tests. When run individually, the test passes.

**Pull request merged**